src/OrleansServiceBus/Providers/Streams/EventHub/EventHubAdapterFactory.cs

```diff
@@ -48,15 +52,30 @@ public void Init(IProviderConfiguration providerConfig, string providerName, Log
 
             logger = log;
             serviceProvider = svcProvider;
+            receivers = new ConcurrentDictionary<QueueId, EventHubAdapterReceiver>();
+
             adapterConfig = new EventHubStreamProviderConfig(providerName);
             adapterConfig.PopulateFromProviderConfig(providerConfig);
-
             hubSettings = adapterConfig.GetEventHubSettings(providerConfig, serviceProvider);
-            checkpointSettings = adapterConfig.GetCheckpointSettings(providerConfig, serviceProvider);
-
-            receivers = new ConcurrentDictionary<QueueId, EventHubAdapterReceiver>();
             client = EventHubClient.CreateFromConnectionString(hubSettings.ConnectionString, hubSettings.Path);
-            bufferPool = new FixedSizeObjectPool<FixedSizeBuffer>(adapterConfig.CacheSizeMb, pool => new FixedSizeBuffer(1 << 20, pool));
+
+            if (CheckpointFactory == null)
+            {
+                checkpointSettings = adapterConfig.GetCheckpointSettings(providerConfig, serviceProvider);
+                CheckpointFactory = partition => EventHubCheckpoint.Create(checkpointSettings, adapterConfig.StreamProviderName, partition);
```

Comment:
I assume the intention is so that in the future we can move the default azure table checkpointer to its own nuget that depends on both azure storage and & service bus, correct? Not saying we should do this now, but I'd like to at some point, especially when we revamp our programmatic config story.